### PR TITLE
v12 release 

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

This is feature branch for merging particular v12 features/toggles/changes. It will be merged to `main` branch when v12 is ready to release.

- [x] enable changes hidden behind `useBreakingChanges` flag #1313 
- [x] make `includeHiddenElements: false` by default #1317
- [x] v12 migration doc 

Issues & fixes:
- [x] #1338 / https://github.com/testing-library/jest-native/pull/151
- [x] #1350 / #1352 
- [x] #1351 / #1355 

### Test plan

All checks should pass.

### CODE REVIEW

**Comment by thymikee** [comment]
Do we have a release plan already? Starting with some RC or going straight to the stable release channel?

**Comment by pierrezimmermannbam** [comment]
I think a RC is a good idea, I'll also try the latest version on my project

**Comment by pierrezimmermannbam** [comment]
I just tried it on my current project, it doesn't work with `toBeOnTheScreen` matcher, I opened an issue #1338

**Comment by mdjastrzebski** [comment]
@pierrezimmermannbam Jest Native 5.4.2 is out with the fix!

**Comment by mdjastrzebski** [comment]
@pierrezimmermannbam how about other issues beside `toBeOnTheScreen()` issue? Where there any or all green?

**Comment by pierrezimmermannbam** [comment]
@mdjastrzebski I'll test with newer version of jest-native because it may have hidden other issues but so far I haven't seen other issues

**Comment by pierrezimmermannbam** [comment]
No other issues on my project, I've tested it on an older project as well, just got some byRole queries not working because they targeted non accessible elements  🎉

**Comment by AugustinLF** [comment]
Back from vacation, I'll try to update, will keep you updated in the upcoming days :)

**Comment by mdjastrzebski** [comment]
I've just released [RC 1](https://github.com/callstack/react-native-testing-library/releases/tag/v12.0.0-rc.1) with fix for #1350.

**Comment by mdjastrzebski** [comment]
Just released [RC 2](https://github.com/callstack/react-native-testing-library/releases/tag/v12.0.0-rc.2) 🎉

**Comment by mdjastrzebski** [comment]
@AugustinLF, @pierrezimmermannbam, @MattAgn the RC 2 seems to be ready from my perspective, all reported issues (#1338, ) have been fixed. 

Do you have any more tests to perform? Are you aware of any more issues?

File: website/docs/MigrationV12.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// No need to "accessible" prop for Text, as it is implicitly accessible element.
+<Text role="button">Button</Text>
+```
+
+While following elements will not match:
+
+```ts
+// Missing "accessible" prop for View
+<View accessible={false} role="button" />
```
Comment (thymikee) [inline website/docs/MigrationV12.md:41]:
```suggestion
<View role="button" />
```

File: website/docs/MigrationV12.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// Explicit "accessible={false}" prop for View
+<View accessible={false} role="button" />
+
+// Explicit "accessible={false}" for Text,which is implicitly accessible element
```
Comment (thymikee) [inline website/docs/MigrationV12.md:46]:
```suggestion
// Explicit "accessible={false}" for Text, which is implicitly accessible element
```

**Review by thymikee — APPROVED** [review]

**Review by AugustinLF — APPROVED** [review]

**Comment by mdjastrzebski** [comment]
If there are no more issues and/or comments regarding this RC, let's plan release for next Tuesday, Mar 14 😄

**Pull request merged**